Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/MicrosoftDefenderAdvancedThreatProtection.py

```diff
@@ -2559,7 +2576,273 @@ def get_indicator_dbot_object(indicator):
+    Args:
+        args: the arguments required to the command being called, under cmd
+        cmd: the command to schedule by after the current command
+        upload_function: the function that initiates the uploading to the API
+        results_function: the function that retrieves the status of the previously initiated upload process
+        uploaded_item: the type of item being uploaded
+
+    Returns:
+
+    """
+    ScheduledCommand.raise_error_if_not_supported()
+    interval_in_secs = int(args.get('interval_in_seconds', 5))
+    # distinguish between the initial run, which is the upload run, and the results run
+    is_first_run = 'machine_action_id' not in args
+    demisto.debug(f'polling args: {args}')
+    if is_first_run:
+        command_results = action_func(client, args)
+        outputs = command_results.outputs
+        # schedule next poll
+        polling_args = {
+            'machine_action_id': outputs.get('action_id'),
+            'interval_in_seconds': interval_in_secs,
+            'polling': True,
+            **args,
+        }
+        scheduled_command = ScheduledCommand(
+            command=cmd,
+            next_run_in_seconds=interval_in_secs,
+            args=polling_args,
+            timeout_in_seconds=600)
+        command_results.scheduled_command = scheduled_command
+        return command_results
+
+    # not a first run
+
+    command_result = results_function(client, args)
+    action_status = command_result.outputs.get("status")
+    command_status = command_result.outputs.get("commands", [])[0].get("commandStatus")
+    if action_status == 'Failed' or command_status == 'Failed':
+        raise Exception(f'Command failed to get results. {command_result.outputs.get("commands", [])[0].get("errors")}')
+    elif command_status != 'Completed' or action_status == 'InProgress':
+        # schedule next poll
+        polling_args = {
+            'interval_in_seconds': interval_in_secs,
+            'polling': True,
+            **args
+        }
+
+        scheduled_command = ScheduledCommand(
+            command=cmd,
+            next_run_in_seconds=interval_in_secs,
+            args=polling_args,
+            timeout_in_seconds=600)
+
+        command_result = CommandResults(scheduled_command=scheduled_command)
+        return command_result
+
+    # action was completed
+    else:
+        return post_polling_process(client, command_result.outputs)
+
+
+def get_live_response_result_command(client, args):
+    machine_action_id = args['machine_action_id']
+    command_index = arg_to_number(args['command_index'])
+    res = client.get_live_response_result(machine_action_id, command_index)
+    file_link = res['value']
+
+    # download link, create file result
+    f_data = client.download_file(file_link)
+    try:
+        outputs = f_data.json()
+    except Exception:
+        outputs = {'value': file_link}
+
+    demisto.results(fileResult('Response Result', f_data.content))
```

Comment:
please return it as part of a `list` along with the other `CommandResults` instead of using `demisto.results`.